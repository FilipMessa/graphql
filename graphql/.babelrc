{
  "env": {
    "test": {
      "plugins": [
        "transform-es2015-modules-commonjs" // https://facebook.github.io/jest/docs/webpack.html#using-with-webpack-2
      ]
    }
  },
  "presets": [
    [
      "env",
      {
        "modules": false,
        "exclude": [
          "transform-regenerator",
          "transform-async-to-generator"
        ],
        "targets": {
          "node": "6.10.2"
        }
      }
    ],
    "flow"
  ],
  "plugins": [
    "fast-async",
    "transform-object-rest-spread",
    "transform-class-properties"
  ]
}
