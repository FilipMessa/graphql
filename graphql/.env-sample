OPTICS_API_KEY=
BOOKING_COM_TOKEN=
ALGOLIA_APP_ID=
ALGOLIA_API_KEY=
