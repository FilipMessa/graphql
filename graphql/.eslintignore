# /node_modules/* and /bower_components/* are ignored by default

config/stub/*
config/webpack.*.js
coverage/*
dist/*
flow-typed/*
