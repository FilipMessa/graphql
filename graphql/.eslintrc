{
  "env": {
    "node": true,
    "jest": true,
  },
  "extends": [
    "airbnb-base",
    "prettier",
    "prettier/react",
    "plugin:flowtype/recommended",
  ],
  "parser": "babel-eslint",
  "plugins": [
    "import",
    "prettier",
    "promise",
    "flowtype",
  ],
  "rules": {
    "no-console": 0, # Console needed for serverless debugging.
    "no-use-before-define": 0, # I like the logic going from highlevel to lowlevel.
    "no-shadow": 0, # Shadowing is a nice language feature. Naming is hard.
    "no-nested-ternary": 0, # Helps brevity of the code.
    "import/prefer-default-export": 0, # Keep imports of helper functions consistent.
    "prettier/prettier": [
      "error",
      {
        "bracketSpacing": true,
        "printWidth": 80,
        "singleQuote": true,
        "tabWidth": 2,
        "trailingComma": "all",
      }
    ],
    "promise/always-return": "error",
    "promise/no-return-wrap": "error",
    "promise/param-names": "error",
    "promise/catch-or-return": "error",
    "promise/no-native": "off",
    "promise/no-nesting": "warn",
    "promise/no-promise-in-callback": "warn",
    "promise/no-callback-in-promise": "warn",
    "promise/avoid-new": "off",
    "promise/prefer-await-to-then": "error",
    "promise/prefer-await-to-callbacks": "off",
    "flowtype/require-valid-file-annotation": [
      "warn",
      "always", {
        "annotationStyle": "line"
      }
    ]
  }
}
