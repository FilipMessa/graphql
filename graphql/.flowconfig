[ignore]
.*/dist/.*

[include]

[libs]

[options]
module.name_mapper='^express-graphql$' -> '<PROJECT_ROOT>/config/stub/GeneralStub.js'
module.name_mapper='^graphql-iso-date$' -> '<PROJECT_ROOT>/config/stub/GeneralStub.js'
