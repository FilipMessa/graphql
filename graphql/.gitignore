/.serverless/
/coverage/
/dist/
/node_modules/
.env
