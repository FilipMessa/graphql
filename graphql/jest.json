{
  "collectCoverage": true,
  "collectCoverageFrom": [
    "src/**.js"
  ],
  "coverageReporters": [
    "lcov",
    "text",
    "text-summary"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 80,
      "functions": 80,
      "lines": 80,
      "statements": 80
    }
  },
  "logHeapUsage": true,
  "verbose": false
}
