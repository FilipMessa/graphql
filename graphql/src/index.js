import graphqlHTTP from 'express-graphql';
import Schema from './Schema';
import { createContext } from './services/GraphqlContext';
import Logger from './services/Logger';

process.on('unhandledRejection', reason => {
  Logger.error(reason);
});

exports.default = (request: Object, response: Object) => {
  const token = request.headers.authorization || null;

  return graphqlHTTP({
    schema: Schema,
    pretty: true,
    graphiql: true,
    context: createContext(token),
    formatError(error) {
      Logger.error(`${error.name}: ${error.message}`);
      return error;
    },
  })(request, response);
};
