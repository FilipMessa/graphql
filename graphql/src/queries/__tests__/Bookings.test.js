import { graphql } from 'graphql';
import schema from '../../Schema';

describe('arrival query', () => {
  it('should return valid id field', async () => {
    const arrivalQuery = `{
        allBookings {
          arrival {
            id
          }
        }
      }`;
    const result = await graphql(schema, arrivalQuery);
    expect(result).toMatchSnapshot();
  });
});

describe('departure query', () => {
  it('should return valid id field', async () => {
    const departureQuery = `{
        allBookings {
          departure {
            id
          }
        }
      }`;
    const result = await graphql(schema, departureQuery);
    expect(result).toMatchSnapshot();
  });
});
