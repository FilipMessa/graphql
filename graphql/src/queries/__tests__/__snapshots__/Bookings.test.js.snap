// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrival query should return valid id field 1`] = `
Object {
  "data": Object {
    "allBookings": Array [
      Object {
        "arrival": Object {
          "id": "{\\"where\\":{\\"code\\":\\"CDG\\",\\"name\\":\\"Paris\\"},\\"when\\":0}",
        },
      },
      Object {
        "arrival": Object {
          "id": "{\\"where\\":{\\"code\\":\\"DXB\\",\\"name\\":\\"Dubai\\"},\\"when\\":0}",
        },
      },
      Object {
        "arrival": Object {
          "id": "{\\"where\\":{\\"code\\":\\"KUF\\",\\"name\\":\\"Samara\\"},\\"when\\":0}",
        },
      },
    ],
  },
}
`;

exports[`departure query should return valid id field 1`] = `
Object {
  "data": Object {
    "allBookings": Array [
      Object {
        "departure": Object {
          "id": "{\\"where\\":{\\"code\\":\\"LGW\\",\\"name\\":\\"London\\"},\\"when\\":0}",
        },
      },
      Object {
        "departure": Object {
          "id": "{\\"where\\":{\\"code\\":\\"STN\\",\\"name\\":\\"London\\"},\\"when\\":0}",
        },
      },
      Object {
        "departure": Object {
          "id": "{\\"where\\":{\\"code\\":\\"PRG\\",\\"name\\":\\"Prague\\"},\\"when\\":0}",
        },
      },
    ],
  },
}
`;
